server/tests/helpers/utils.js

```diff
@@ -34,61 +29,12 @@ export const verifyUser = async (agent, data) => {
   return loginUser(agent, data);
 };
 
-export const getCategoryId = async (values) => {
-  try {
-    const [category] = await Category.findOrCreate({
-      where: { name: values },
-      defaults: {
-        name: values
-      }
-    });
-    const { id } = category;
-    return id;
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-export const getUserId = async (email, username) => {
-  try {
-    const user = await User.create(
-      getUserData({
-        email,
-        username
-      })
-    );
-    const { id } = user;
-    return id;
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-export const getTagId = async (values) => {
-  try {
-    const [tag] = await Tag.findOrCreate({
-      where: { name: values },
-      defaults: {
-        name: values
-      }
-    });
-    const { id } = tag;
-    return id;
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-export const getArticleId = async (title, body, userId) => {
+export const getModelObjectId = async (model, obj) => {
   try {
-    const categoryId = await getCategoryId('technology');
-    const newArticle = await Article.create({
-      title,
-      body,
-      userId,
-      categoryId
+    const [modelObject] = await model.findOrCreate({
```

Comment:
why are you using `findOrCreate`. Surely you want to use a method specific to find, we do not want the Id created if it doesn't exist except that is your aim.

Reply:
I refactored about 3 functions into one. They looked for a particular item but if it’s not there creates it.